src/openforms/forms/api/serializers/logic/validators.py

```diff
@@ -0,0 +1,25 @@
+from uuid import UUID
+
+from django.core.exceptions import ObjectDoesNotExist
+from django.utils.translation import gettext_lazy as _
+
+from rest_framework import serializers
+
+from openforms.forms.models import FormStep
+
+
+class FormStepJsonLogicActionValidator:
+    """
+    Validate that a form step referenced in the action of a JSON Logic rule exists
+    """
+
+    def __call__(self, step_uuid: UUID):
+        if step_uuid == "":
+            return
+
+        try:
+            FormStep.objects.get(uuid=step_uuid)
+        except ObjectDoesNotExist:
+            raise serializers.ValidationError(
+                _("Invalid form step specified in logic action."), code="invalid"
+            )
```

Comment:
this will be a performance nightmare with the bulk endpoint - can you look into the custom field that fetches data from the serializer context and use a similar approach (separate commit please, so we can still backport the 'naive' fix).